### PR TITLE
Validation of credentials done

### PR DESCRIPTION
**Pull request by dreadpool2**
*validation of password and email done
*Utils file made
*AuthUtils made
*LoginActivityViewModel made
*required files indented

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/LoginActivity.java [hunk 1]
```diff
@@ -53,9 +62,14 @@ protected void onCreate(Bundle savedInstanceState) {
         progressDialog.setMessage("Logging you in...");
 
         loginBtn.setOnClickListener(v -> {
-            //loginUser(ei.getText().toString(),e2.getText().toString());
-            loginUser("hey@hey.hey", "heyheyhey");
-            progressDialog.show();
+            //String email = usernameET.getText().toString();
+            //String password = passwordET.getText().toString();
+            String email = "hey@hey.hey";
+            String password = "heyheyhey";
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/LoginActivity.java]: Don't hardcode. Take the input from the user

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -12,10 +12,17 @@
...
     <string name="login">Login</string>
     <string name="sign_in">No account yet? Create one</string>
     <string name="email">example@gmail.com</string>
 
+    <!--auth details-->
+    <string name="error_email_required">Email address is required</string>
+    <string name="error_enter_valid_email">Enter valid email adress</string>
+    <string name="error_password_required">Password is required</string>
+
+
```
Comment (nikit19) [inline app/src/main/res/values/strings.xml]: Please avoid  blank lines here and at other places where they are not required

File: app/src/main/java/org/fossasia/openevent/general/LoginActivity.java [hunk 3]
```diff
@@ -92,6 +106,34 @@ public void loginUser(String title, String body) {
                 }));
     }
 
+
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/LoginActivity.java]: Please avoid blank lines here and at other places where they are not required

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -12,10 +12,17 @@
     <string name="event_card_date">Event Date :</string>
 
     <!--profile details-->
+    <string name="email_head">Email</string>
```
Comment (nikit19) [inline app/src/main/res/values/strings.xml]: email_head ?

File: app/src/main/java/org/fossasia/openevent/general/LoginActivity.java [hunk 5]
```diff
@@ -92,6 +106,34 @@ public void loginUser(String title, String body) {
...
+        } else if (!Utils.isEmailValid(email)) {
+            handleError(emailWrapper, R.string.error_enter_valid_email);
+            return false;
+        }
+
+        if (isEmpty(password)) {
+            handleError(passwordWrapper, R.string.error_password_required);
+            return false;
+        }
+
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/LoginActivity.java]: Please avoid blank lines here and at other places where they are not required

File: app/src/main/java/org/fossasia/openevent/general/LoginActivity.java [hunk 6]
```diff
@@ -92,6 +106,34 @@ public void loginUser(String title, String body) {
                 }));
     }
 
+
+    private boolean validateCredentials(String email, String password) {
+
+        // Reset errors.
+        emailWrapper.setError(null);
+        passwordWrapper.setError(null);
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/LoginActivity.java]: why?

**Review by nikit19 — CHANGES_REQUESTED** [review]
Please see my suggestions

File: app/src/main/java/org/fossasia/openevent/general/LoginActivity.java (hunk 6 above)
Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/LoginActivity.java]: set error method is used to set the error message. just when validate credentials is called it is initialised to null and then values are set

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (dreadpool2) [reply app/src/main/res/values/strings.xml]: the email hint actually. will change the name asap

**Comment by iamareebjamal** [comment]
@nikit19 @dreadpool2 Please create an issue to use kotlin instead and further convert project to kotlin

File: app/src/main/java/org/fossasia/openevent/general/LoginActivity.java (hunk 6 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/LoginActivity.java]: Move this to presenter logic or ViewModel. Keep views free of logic

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/LoginActivity.java]: Okay :+1:

**Comment by dreadpool2** [comment]
@iamareebjamal Please review

**Pull request closed without merging**